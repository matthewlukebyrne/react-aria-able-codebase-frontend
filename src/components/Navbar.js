import styled from 'styled-components';

// Using React Router DOM we can use the "Link" atrribute to switch routing on the href
import { Link } from 'react-router-dom';


const Navbar = () => {
  return (
    <StyledNavbar data-testid='navtest-id'>
      <h1>
        <Link to='/'>Aria-Able</Link>
      </h1>
      <ul>
        <li>
          <Link to='/components'>Components</Link>
        </li>
        <li>
          <Link to='/contact'>Contact</Link>
        </li>
      </ul>
    </StyledNavbar>
  );
};

export default Navbar;

// styled components
const StyledNavbar = styled.nav`
  min-height: 10vh;
  display: flex;
  margin: auto;
  justify-content: space-between;
  align-items: center;
  padding: 0rem 5rem;
  background-color: whitesmoke;
  position: sticky;
  top: 0;
  z-index: 10; // Navbar always stays forward!

  a {
    color: #1b1b1b;
    text-decoration: none;
    font-size: 3rem;
  }
  ul {
    display: flex;
    list-style: none;
  }
  /* Main logo */
  /* #main_logo {
		color: white;
		font-size: 3rem;
		font-weight: lighter;
	} */
  /* List Elements */
  li {
    padding-left: 5rem;
    position: relative;
  }

  // Added Media Queries
  @media (max-width: 1300px) {
    flex-direction: column;
    padding: 2rem;
    #main_logo {
      display: inline-block;
      margin: 1rem;
    }
    ul {
      padding: 2rem 1rem;
      justify-content: space-evenly;
      width: 100%;
      li {
        padding: 0;
      }
    }
  }
`;

// Added active line
// const Line = styled(motion.div)`
// 	height: 0.2rem;
// 	background: #e74e16;
// 	width: 0%;
// 	position: relative;
// 	bottom: -10%;
// 	left: 0%;
// 	@media (max-width: 1300px) {
// 		left: 0%;
// 	}
// `;
