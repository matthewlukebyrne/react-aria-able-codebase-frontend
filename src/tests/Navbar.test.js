import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Navbar from '../components/Navbar';

// TDD - Red Green Refactor
describe('Navbar Section Component Unit Tests', () => {
  // the first test
  it('render the navbar component correctly without crashing with 3 items', () => {
    render(<Navbar />);
    const navBarPonent = screen.getByTestId('navtest-id');
    // make assertions
    expect(navBarPonent).toBeInTheDocument();
    expect(navBarPonent).toHaveTextContent('Components', 'Gallery', 'Contact');
  });

  test('navbar section styled components', () => {
    const navStyles = renderer.create(<Navbar />).toJSON();
    expect(navStyles).toMatchSnapshot();
    expect(navStyles).toHaveStyleRule('min-height', '10vh');
    expect(navStyles).toHaveStyleRule('justify-content', 'space-between');
    expect(navStyles).toHaveStyleRule('background-color', 'whitesmoke');
  });
});
