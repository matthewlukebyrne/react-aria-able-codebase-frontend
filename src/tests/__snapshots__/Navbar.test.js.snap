// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Section Component Unit Tests navbar section styled components 1`] = `
.c0 {
  min-height: 10vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: auto;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0rem 5rem;
  background-color: whitesmoke;
  position: -webkit-sticky;
  position: sticky;
  top: 0;
  z-index: 10;
}

.c0 a {
  color: #1b1b1b;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.5rem;
}

.c0 ul {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  list-style: none;
}

.c0 li {
  padding-left: 5rem;
  position: relative;
}

@media (max-width:1300px) {
  .c0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    padding: 2rem;
  }

  .c0 #main_logo {
    display: inline-block;
    margin: 1rem;
  }

  .c0 ul {
    padding: 2rem 1rem;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
    width: 100%;
  }

  .c0 ul li {
    padding: 0;
  }
}

<nav
  className="c0"
  data-testid="navtest-id"
>
  <h1>
    <a
      href="a"
    >
       
    </a>
    Aria-Able
  </h1>
  <ul>
    <li>
      <a
        href="a"
      >
        Components
      </a>
    </li>
    <li>
      <a
        href="a"
      >
        Gallery
      </a>
    </li>
    <li>
      <a
        href="a"
      >
        Contact
      </a>
    </li>
  </ul>
</nav>
`;
